### PR TITLE
Set webhook matchPolicy to Exact

### LINKED ISSUE #4270: Elasticsearch v1beta1 webhook fires for v1 resources
Attempting to increase the storage size in the volume claim template gets rejected by the webhook. From the operator logs, it appears that both webhook endpoints get triggered for the request. The `v1beta1` endpoint obviously rejects the request while the `v1` endpoint allows it. Somehow kubectl gets the response from the `v1beta1` webhook and the operation gets cancelled. 

```
{"log.level":"debug","@timestamp":"2021-02-24T10:58:37.669Z","log.logger":"controller-runtime.webhook.webhooks","message":"received request","service.version":"1.4.0+4aff0b98","service.type":"eck","ecs.version":"1.4.0","webhook":"/validate-elasticsearch-k8s-elastic-co-v1-elasticsearch","UID":"c991a607-cbee-4335-901b-d81fd277f3f2","kind":"elasticsearch.k8s.elastic.co/v1, Kind=Elasticsearch","resource":{"group":"elasticsearch.k8s.elastic.co","version":"v1","resource":"elasticsearches"}}
{"log.level":"debug","@timestamp":"2021-02-24T10:58:37.669Z","log.logger":"es-validation","message":"validate update","service.version":"1.4.0+4aff0b98","service.type":"eck","ecs.version":"1.4.0","name":"hulk"}
{"log.level":"debug","@timestamp":"2021-02-24T10:58:37.699Z","log.logger":"controller-runtime.webhook.webhooks","message":"received request","service.version":"1.4.0+4aff0b98","service.type":"eck","ecs.version":"1.4.0","webhook":"/validate-elasticsearch-k8s-elastic-co-v1beta1-elasticsearch","UID":"3ce5e7aa-986e-430b-9eee-b9077c7b4dbf","kind":"elasticsearch.k8s.elastic.co/v1beta1, Kind=Elasticsearch","resource":{"group":"elasticsearch.k8s.elastic.co","version":"v1beta1","resource":"elasticsearches"}}
{"log.level":"debug","@timestamp":"2021-02-24T10:58:37.705Z","log.logger":"es-validation","message":"validate update","service.version":"1.4.0+4aff0b98","service.type":"eck","ecs.version":"1.4.0","name":"hulk"}
{"log.level":"debug","@timestamp":"2021-02-24T10:58:37.705Z","log.logger":"controller-runtime.webhook.webhooks","message":"wrote response","service.version":"1.4.0+4aff0b98","service.type":"eck","ecs.version":"1.4.0","webhook":"/validate-elasticsearch-k8s-elastic-co-v1beta1-elasticsearch","code":403,"reason":"Elasticsearch.elasticsearch.k8s.elastic.co \"hulk\" is invalid: [spec.nodeSet[0].volumeClaimTemplates: Invalid value: []v1.PersistentVolumeClaim{v1.PersistentVolumeClaim{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, ObjectMeta:v1.ObjectMeta{Name:\"elasticsearch-data\", GenerateName:\"\", Namespace:\"\", SelfLink:\"\", UID:\"\", ResourceVersion:\"\", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:\"\", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v1.PersistentVolumeClaimSpec{AccessModes:[]v1.PersistentVolumeAccessMode{\"ReadWriteOnce\"}, Selector:(*v1.LabelSelector)(nil), Resources:v1.ResourceRequirements{Limits:v1.ResourceList(nil), Requests:v1.ResourceList{\"storage\":resource.Quantity{i:resource.int64Amount{value:32212254720, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:\"30Gi\", Format:\"BinarySI\"}}}, VolumeName:\"\", StorageClassName:(*string)(0xc000aa48d0), VolumeMode:(*v1.PersistentVolumeMode)(nil), DataSource:(*v1.TypedLocalObjectReference)(nil)}, Status:v1.PersistentVolumeClaimStatus{Phase:\"\", AccessModes:[]v1.PersistentVolumeAccessMode(nil), Capacity:v1.ResourceList(nil), Conditions:[]v1.PersistentVolumeClaimCondition(nil)}}}: Volume claim templates cannot be modified, spec.nodeSet[1].volumeClaimTemplates: Invalid value: []v1.PersistentVolumeClaim{v1.PersistentVolumeClaim{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, ObjectMeta:v1.ObjectMeta{Name:\"elasticsearch-data\", GenerateName:\"\", Namespace:\"\", SelfLink:\"\", UID:\"\", ResourceVersion:\"\", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:\"\", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v1.PersistentVolumeClaimSpec{AccessModes:[]v1.PersistentVolumeAccessMode{\"ReadWriteOnce\"}, Selector:(*v1.LabelSelector)(nil), Resources:v1.ResourceRequirements{Limits:v1.ResourceList(nil), Requests:v1.ResourceList{\"storage\":resource.Quantity{i:resource.int64Amount{value:32212254720, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:\"30Gi\", Format:\"BinarySI\"}}}, VolumeName:\"\", StorageClassName:(*string)(0xc000aa4a60), VolumeMode:(*v1.PersistentVolumeMode)(nil), DataSource:(*v1.TypedLocalObjectReference)(nil)}, Status:v1.PersistentVolumeClaimStatus{Phase:\"\", AccessModes:[]v1.PersistentVolumeAccessMode(nil), Capacity:v1.ResourceList(nil), Conditions:[]v1.PersistentVolumeClaimCondition(nil)}}}: Volume claim templates cannot be modified, spec.nodeSet[2].volumeClaimTemplates: Invalid value: []v1.PersistentVolumeClaim{v1.PersistentVolumeClaim{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, ObjectMeta:v1.ObjectMeta{Name:\"elasticsearch-data\", GenerateName:\"\", Namespace:\"\", SelfLink:\"\", UID:\"\", ResourceVersion:\"\", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:\"\", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v1.PersistentVolumeClaimSpec{AccessModes:[]v1.PersistentVolumeAccessMode{\"ReadWriteOnce\"}, Selector:(*v1.LabelSelector)(nil), Resources:v1.ResourceRequirements{Limits:v1.ResourceList(nil), Requests:v1.ResourceList{\"storage\":resource.Quantity{i:resource.int64Amount{value:32212254720, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:\"30Gi\", Format:\"BinarySI\"}}}, VolumeName:\"\", StorageClassName:(*string)(0xc000aa4ac0), VolumeMode:(*v1.PersistentVolumeMode)(nil), DataSource:(*v1.TypedLocalObjectReference)(nil)}, Status:v1.PersistentVolumeClaimStatus{Phase:\"\", AccessModes:[]v1.PersistentVolumeAccessMode(nil), Capacity:v1.ResourceList(nil), Conditions:[]v1.PersistentVolumeClaimCondition(nil)}}}: Volume claim templates cannot be modified]","UID":"3ce5e7aa-986e-430b-9eee-b9077c7b4dbf","allowed":false}
{"log.level":"debug","@timestamp":"2021-02-24T10:58:37.870Z","log.logger":"controller-runtime.webhook.webhooks","message":"wrote response","service.version":"1.4.0+4aff0b98","service.type":"eck","ecs.version":"1.4.0","webhook":"/validate-elasticsearch-k8s-elastic-co-v1-elasticsearch","code":200,"reason":"","UID":"c991a607-cbee-4335-901b-d81fd277f3f2","allowed":true}
{"log.level":"debug","@timestamp":"2021-02-24T10:58:37.883Z","log.logger":"controller-runtime.webhook.webhooks","message":"received request","service.version":"1.4.0+4aff0b98","service.type":"eck","ecs.version":"1.4.0","webhook":"/validate-elasticsearch-k8s-elastic-co-v1beta1-elasticsearch","UID":"7a6fba0f-7133-4efa-8e0e-9c88623081ac","kind":"elasticsearch.k8s.elastic.co/v1beta1, Kind=Elasticsearch","resource":{"group":"elasticsearch.k8s.elastic.co","version":"v1beta1","resource":"elasticsearches"}}
```

### PR DESCRIPTION
**Pull request by charith-elastic**
`admissionregistration.k8s.io/v1` changed the default `matchPolicy` to `Equivalent` from `Exact`. This results in both `v1` and `v1beta1` webhooks firing for resources like Elasticsearch, Kibana and APMServer. 

Fixes #4270

### CODE REVIEW

**Comment by charith-elastic** [comment]
Jenkins test this please

File: config/webhook/manifests.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ webhooks:
       namespace: system
       path: /validate-agent-k8s-elastic-co-v1alpha1-agent
   failurePolicy: Ignore
+  matchPolicy: Exact
```
Comment (pebrc) [inline config/webhook/manifests.yaml:18]: What happens on k8s versions < 1.15 where I believe this property did not exist yet?

Comment (charith-elastic) [reply config/webhook/manifests.yaml:18]: Good catch. I should have checked the versions where it is supported.

**Comment by charith-elastic** [comment]
Jenkins test this please

**Review by pebrc — COMMENTED** [review]
LGTM code-wise.  

Just remind me what we do in the `all-in-one.yaml`, am I right that we are assuming `kubeVersion` 1.12.0? 

If that is correct: won't users that install via the `all-in-one.yaml` manifest still run into this issue if they are on a recent k8s version?

Also: you mentioned in the original issue that only Helm installs would be affected? I am somehow missing a puzzle piece to fully understand the reason why.

**Comment by charith-elastic** [comment]
When installing with Helm, the manifests are generated based on the version of Kubernetes running on the destination cluster. So, if it's 1.16+, the webhook API version gets set to `v1` and additional fields like `sideEffect`, `admissionReviewVersions` are set as well.

For `all-in-one.yaml`, the Kubernetes version is explicitly set to 1.12.0. The resulting manifest does not contain any of the above modifications. (API version is set to `v1beta1` and `sideEffect` and `admissionReviewVersions` are not set.)

The webhook `matchPolicy` used to default to `Exact` in webhook `v1beta1` API. They changed it to `Equivalent` in the `v1` API. People who installed using Helm on 1.16+ would have the `v1` webhook and get the new match policy. Those who installed using the `all-in-one.yaml`  would have the `v1beta1` webhook and the match policy will be the old default -- which is what we want.

**Review by pebrc — APPROVED** [review]

**Pull request merged**